interfaces/dbus/spec.go

```diff
@@ -32,6 +34,94 @@ type Specification struct {
 	// Snippets are indexed by security tag.
 	snippets     map[string][]string
 	securityTags []string
+
+	sessionServices map[string]*Service
+	systemServices  map[string]*Service
+}
+
+// Service describes an activatable D-Bus service
+type Service struct {
+	SecurityTag string
+	BusName     string
+	Content     []byte
+}
+
+// AddService adds a new D-Bus service
+func (spec *Specification) AddService(bus, name string, appInfo *snap.AppInfo) error {
+	if !appInfo.IsService() {
+		return fmt.Errorf("activatable D-Bus services must be daemons")
+	}
+
+	serviceTemplate := `[D-BUS Service]
```

Comment:
Could you add a code comment that references the specification of this format please.